installer/pkg/config-generator/generator.go

```diff
@@ -61,6 +63,78 @@ func New(cluster config.Cluster) ConfigGenerator {
 	}
 }
 
+// maoOperatorConfig contains configuration for mao managed stack
+// TODO(enxebre): move up to "github.com/coreos/tectonic-config
+type maoOperatorConfig struct {
+	metav1.TypeMeta `json:",inline"`
+	Provider        string         `json:"provider"`
+	AWS             *awsConfig     `json:"aws"`
+	Libvirt         *libvirtConfig `json:"libvirt"`
+}
+
+type libvirtConfig struct {
+	ClusterName string `json:"clusterName"`
+	URI         string `json:"uri"`
+	NetworkName string `json:"networkName"`
+	Replicas    int    `json:"replicas"`
+}
+
+type awsConfig struct {
+	ClusterName      string `json:"clusterName"`
+	ClusterID        string `json:"clusterID"`
+	Region           string `json:"region"`
+	AvailabilityZone string `json:"availabilityZone"`
+	Image            string `json:"image"`
+	Replicas         int    `json:"replicas"`
+}
+
+func (c *ConfigGenerator) maoConfig() *maoOperatorConfig {
+	cfg := maoOperatorConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "machineAPIOperatorConfig",
+		},
+	}
+
+	cfg.Provider = tectonicCloudProvider(c.Platform)
+
+	switch c.Platform {
+	case config.PlatformAWS:
+		// TODO: Use actual RHCOS channel.
+		ami, err := rhcos.AMI("tested", c.Region)
+		if err != nil {
+			log.Fatalf("Failed to lookup RHCOS AMI: %v", err)
+			return nil
+		}
+
+		if len(c.AWS.EC2AMIOverride) > 0 {
```

Comment:
The following is easier to read and performs better (no need to lookup the AMI if there is an override):

```go
if c.AWS.EC2AMIOverride != "" {
    ami = c.AWS.EC2AMIOverride
} else {
    ami, err := rhcos.AMI("tested", c.Region)
		if err != nil {
			log.Fatalf("Failed to lookup RHCOS AMI: %v", err)
			return nil
		}   
}
```

Reply:
> ```go
> log.Fatalf("Failed to lookup RHCOS AMI: %v", err)
> ```

+1 to the approach @crawford recommends, but I'd rather update `maoConfig` to allow returning an error and then return the AMI-lookup error unwrapped here.  If the errors from the lookup helper are not specific enough, we should update them, instead of requiring callers to wrap them with more context.

Reply:
Yeah, I originally did it this way to help surface errors resulting from the fact that the AMI isn't really available. I agree that this method should return an error. I will fix it when I rebase for ultimate performance.

Reply:
just don't copy his code and accidentally alias the ami variable   :)

Reply:
Go sucks.

Reply:
Fixed.